rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,89 @@
+
+### Registering nodes
+
+A new sub-kind to the `node` resource will be added for registered OpenSSH nodes: `openssh`. The absence of a `node` resource sub-kind will imply that a node is a Teleport agent node, making this change backwards compatible.
+
+#### Registering with `tctl`
+
+OpenSSH nodes can already be manually registered using `tctl create --force /path/to/node.yml`, though some changes should be made to make the process more straightforward for users. Agentless EC2 discovery mode will registered discovered OpenSSH nodes without needing user intervention, but if automatically registering an OpenSSH node fails a user may want to register a node manually. Currently nodes cannot be created with `tctl`, only upserted. This limitation should be removed so users can create nodes without having to pass `--force` to `tctl create`.
+
+Furthermore, `tctl` should not require as many fields to be set when creating nodes. This is an example `node` resource that will work with `tctl create` today:
+
+```yaml
+kind: node
+metadata:
+  name: 5da56852-2adb-4540-a37c-80790203f6a9
+spec:
+  addr: 1.2.3.4:22
+  hostname: agentless-node
+version: v2
+```
+
+`tctl create` will auto-generate `metadata.name` if it is not already set so users don't have to generate GUIDs themselves. Also, if `sub_kind` is set to `openssh`, `spec.public_addr` will not be allowed for registered OpenSSH nodes as it is not needed.
+
+### Session recording
+
+Currently session recording is required to be set be in `proxy` mode to work with OpenSSH nodes. That is not going to change, but ideally this requirement could be lifted when Teleport agent nodes and registered OpenSSH nodes are both in a single cluster. When establishing an SSH connection inside a cluster, depending on what the session recording mode is set to the appropriate type of session recording would be used:
+
+If session recording is in `node` or `node-sync` mode:
+
+- If the node is a Teleport Agent node, the node would record the session and upload it as normal.
+- If the node is a registered OpenSSH node, the Proxy would terminate and record the SSH session and upload it.
+
+If session recording is in `proxy` or `proxy-sync` mode:
+
+- Behavior would be unaffected. The Proxy would terminate, record and upload the session. This mode will still be required if users of a cluster wish to connect to unregistered OpenSSH nodes.
+
+I propose that `proxy` or `proxy-sync` session recording modes continue to be required when connecting to any OpenSSH node through Teleport, *at first*. When registering OpenSSH nodes and preforming RBAC checks on them is completed and possibly released, then work could be done to streamline session recording with registered OpenSSH nodes.
```

Comment:
Can you move this section at the end and mark it as "possible extensions". Currently looks like something that we're going to implement as a part of this RFD which is not true.

Reply:
Done